Library/Homebrew/extend/os/mac/diagnostic.rb

```diff
@@ -438,6 +439,36 @@ def check_if_supported_sdk_available
           #{update_instructions}
         EOS
       end
+
+      # The CLT 10.x -> 11.x upgrade process on 10.14 contained a bug which broke the SDKs.
+      # Notably, MacOSX10.14.sdk would indirectly symlink to MacOSX10.15.sdk.
+      # This diagnostic was introduced to check for this and recommend a full reinstall.
+      def check_broken_sdks
```

Comment:
Would this not also pickup mismatches on Xcode-only setups? Though, I suppose the standard headline of "ignore this if things are not broken" would apply.

Still, if the warning shows up in the aforementioned case, it might be better to avoid recommending a reinstall for Xcode SDK mismatches, as trying to reinstall Xcode can be a huge pain. Some user may unnecessarily try it given the warning.

Reply:
> Would this not also pickup mismatches on Xcode-only setups?

If they modified their Xcode SDKs for some reason, sure. While there may not be a bug in Xcode installs, I don't see any reason to explicitly exclude checking Xcode. Any unmodified installation will not trigger this diagnostic.

Reply:
Actually, my original comment was from a brief misunderstanding of the test. The test is fine.